gpjax/kernels/stationary/changepoint.py

```diff
@@ -0,0 +1,115 @@
+    self.kernels: A list of exactly two kernels that will be switched.
+    self.tswitch: The point at which to change to a different kernel.
+        for example: if x and y are both less than tswitch, then you would use kernels[0]
+                     if x and y are both greater than or equal to tswitch, then you would use kernels[1]
+                     otherwise return cross-covariance of 0
+    """
+
+    kernels: List[AbstractKernel] = None
+    operator: Callable = static_field(None)
+    tswitch: ScalarFloat = param_field(
+        jnp.array(1.0), bijector=tfb.Identity(), trainable=False
+    )
+
+    def __post_init__(self):
+        # Add kernels to a list, flattening out instances of this class therein, as in GPFlow kernels.
+        kernels_list: List[AbstractKernel] = []
+
+        for kernel in self.kernels:
+            if not isinstance(kernel, AbstractKernel):
+                raise TypeError("can only combine Kernel instances")  # pragma: no cover
+
+            if isinstance(kernel, self.__class__):
+                kernels_list.extend(kernel.kernels)
+            else:
+                kernels_list.append(kernel)
+
+        self.kernels = kernels_list
+
+    def __call__(
+        self,
+        x: Float[Array, " D"],
+        y: Float[Array, " D"],
+    ) -> ScalarFloat:
+        r"""Evaluate the kernel on a pair of inputs.
+
+        Args:
+            x (Float[Array, " D"]): The left hand input of the kernel function.
+            y (Float[Array, " D"]): The right hand input of the kernel function.
+
+        Returns
+        -------
+            ScalarFloat: The evaluated kernel function at the supplied inputs.
+        """
+
+        def get_function_index(x, y, tswitch=1):
+            r"""
+            Specify four possible indices given x, y, and tswitch.
+
+            Args:
+                    x: Left hand argument of kernel function's call
+                    y: Right hand argument of kernel function's call
+                    tswitch: point at which to change to a different kernel
+            """
+            # Four possible indexes: 0, 1, 2, 3
+            indx = 3  # if indx doesn't get set to 0, 1, or 2, then default 3
+            # lessthan means that at tswitch, you are already switched
+            cond1 = jnp.less(x, tswitch)
+            cond2 = jnp.less(y, tswitch)
+            indx = jnp.where(jnp.logical_and(cond1, cond2), 0, indx)
+            indx = jnp.where(jnp.logical_and(jnp.invert(cond1), cond2), 1, indx)
+            indx = jnp.where(jnp.logical_and(cond1, jnp.invert(cond2)), 2, indx)
+            return indx.squeeze().astype("uint8")
+
+        def k_zero(x, y):
+            r"""Return 0 covariance"""
+            out = jnp.float64(0)
+            return out.squeeze()
+
+        indx = get_function_index(x, y, tswitch=self.tswitch)
+
+        flst = [self.kernels[0].__call__, k_zero, k_zero, self.kernels[1].__call__]
+        K = jax.lax.switch(indx, flst, x, y)
```

Comment:
Maybe on naming. I don't know what `flst` means. Maybe something like `kernel_options`. 

Also, why do we need `self.kernels[0].__call__` and `self.kernels[1].__call__`. Does this not work?
```
kernel_options = self.kernels + [k_zero].
```
Would this help with extending this kernel to more than two kernels? Or is there some issue with the computing this indexing?

Reply:
To extend to more than two kernels, I would need to rewrite the `get_function_index()`. 

The technical part is to write a conditional operator that returns an index `indx` that can directly call `self.kernels[indx].__call__`, otherwise return `k_zero`.  I'll think about how to do this.